### PR TITLE
Support Nodeinfo2

### PR DESCRIPTION
**Pull request by annando**
We now support Nodeinfo2, see here: https://git.feneas.org/jaywink/nodeinfo2

This is the successor of Nodeinfo 1 that we already support in version 1.0 and 2.0. (Yes, Nodeinfo 2.0 is different from Nodeinfo2)

### CODE REVIEW

**Comment by annando** [comment]
I don't think that inheritance is the best way here. I think that it would be best to separate the three different versions in three separate module classes and move the common stuff in the already existing `Nodeinfo` model.

File: src/Module/NodeInfo210.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		$nodeinfo['services']['inbound'][]  = 'rss2.0';
+		$nodeinfo['services']['outbound'][] = 'atom1.0';
+
+		if ($imap) {
+			$nodeinfo['services']['inbound'][] = 'imap';
+		}
+
+		header('Content-type: application/json; charset=utf-8');
+		echo json_encode($nodeinfo, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+		exit;
```
Comment (MrPetovan) [inline src/Module/NodeInfo210.php:82]: Please use `System::jsonExit()` instead.

File: src/Module/NodeInfo120.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+		if ($imap) {
+			$nodeinfo['services']['inbound'][] = 'imap';
+		}
+
+		$nodeinfo['metadata']['explicitContent'] = $config->get('system', 'explicit_content', false) == true;
+
+		header('Content-type: application/json; charset=utf-8');
+		echo json_encode($nodeinfo, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+		exit;
```
Comment (MrPetovan) [inline src/Module/NodeInfo120.php:84]: Please use `System::jsonExit()` instead.

File: src/Module/NodeInfo110.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+		if (Addon::isEnabled('twitter')) {
+			$nodeinfo['metadata']['services']['inbound'][] = 'twitter';
+		}
+
+		$nodeinfo['metadata']['explicitContent'] = $config->get('system', 'explicit_content', false) == true;
+
+		header('Content-type: application/json; charset=utf-8');
+		echo json_encode($nodeinfo, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+		exit;
```
Comment (MrPetovan) [inline src/Module/NodeInfo110.php:90]: Please use `System::jsonExit()` instead.

File: src/Core/System.php [hunk 4]
```diff
@@ -136,12 +136,13 @@ public static function jsonError($httpCode, $data, $content_type = 'application/
...
 	 * After finishing the process is getting killed.
 	 *
-	 * @param mixed  $x The input content.
-	 * @param string $content_type Type of the input (Default: 'application/json').
+	 * @param mixed   $x The input content.
+	 * @param string  $content_type Type of the input (Default: 'application/json').
+	 * @param integer $options JSON options
 	 */
-	public static function jsonExit($x, $content_type = 'application/json') {
+	public static function jsonExit($x, $content_type = 'application/json', int $options = null) {
```
Comment (MrPetovan) [inline src/Core/System.php:143]: According to the official function definition, the default value for `$options` should be `0`: https://www.php.net/manual/en/function.json-encode.php

**Pull request merged**